### DISCUSSION TITLE
How to parse a python file with arguments to hammerDB

### DISCUSSION THREAD

**Discussion by tieonster** [TPC-Council/HammerDB#582] [Q&A] [answered]
Hello all,

I am trying to get hammerDB to run a python file with arguments that are accepted with the argparse library in Python. 

I understand that this is the example we use to run a python file from the CLI directly based on the [documentation]([[url](https://www.hammerdb.com/docs/ch09s10.html)]):
./hammerdbcli py auto autorunbuild.py

I want to do something like this:
./hammerdbcli py auto autorunbuild.py --hostname XXX --password YYY
to be able to parse arguments from the command line to the python script directly.

Can I check how I can go about doing this?

**Comment by sm-shaw** [comment, marked as answer]
At present, within HammerDB there isn't the option to parse additional command line arguments aside from the script name:
Parameters are set in your script (or at the command line) using the `diset` command, for example:
```
diset('connection','mssqls_tcp','false')
diset('connection','mssqls_port','1433')
diset('connection','mssqls_azure','false')
diset('connection','mssqls_encrypt_connection','true')
diset('connection','mssqls_trust_server_cert','true')
diset('connection','mssqls_authentication','windows')
diset('connection','mssqls_server','{(local)}')
```
Although I have not looked into it may be possible to use argparse to convert a command line argument such as 
`--connection:mssqls_tcp false` into running the command `diset('connection','mssqls_tcp','false') `. 
To do this, you would first have to modify `hammerdbcli` to accept additional command line arguments.
